### ISSUE TITLE
MacOS failed 1 test in current master

### ISSUE THREAD

**Issue by ulwlu** [pdm-project/pdm#347]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

**Make sure you run commands with `-v` flag before pasting the output.**

## Steps to reproduce

On MacOS, run `pdm run test`.

## Actual behavior

Failed 13 tests.

<img width="1258" alt="ss 4" src="https://user-images.githubusercontent.com/41639488/112740980-708f5f80-8fbc-11eb-8eb5-feb43847898e.png">

![ss 2](https://user-images.githubusercontent.com/41639488/112740983-7422e680-8fbc-11eb-8d6f-931c74c7c3fb.png)

## Expected behavior

Clear all tests.

## Environment Information

```bash
PDM version:        1.4.4
Python Interpreter: /usr/local/opt/python@3.9/bin/python3.9 (3.9.0)
Project Root:       /Users/kohei.murakami/.ghq/github.com/ulwlu/pdm
{
  "implementation_name": "cpython",
  "implementation_version": "3.9.0",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "19.6.0",
  "platform_system": "Darwin",
  "platform_version": "Darwin Kernel Version 19.6.0: Thu Oct 29 22:56:45 PDT 2020; root:xnu-6153.141.2.2~1/RELEASE_X86_64",
  "python_full_version": "3.9.0",
  "platform_python_implementaiton": "CPython",
  "python_version": "3.9",
  "sys_platform": "darwin"
}
```

## Remarks

I think this is related to https://github.com/pdm-project/pdm/pull/109#issuecomment-613786780 .

I'm now trying to fix tests, but would love to know the proposal is correct.

First of all, ` tests/test_utils.py::test_find_python_in_path` is failed because brew-installed-python's `sys.prefix` and `sys.executable` are different. So it always fails.

```
sys.executable: /usr/local/Cellar/python@3.9/3.9.0_1/bin/python3.9

sys.prefix: /usr/local/Cellar/python@3.9/3.9.0_1/Frameworks/Python.framework/Versions/3.9/bin/python3
```

Will it be fine if I change this test into like "find the sys.executable path up to version just like `/usr/local/Cellar/python@3.9`, and compare it to sys.prefix with `startwith`" if MacOS?

**Comment by frostming** [comment]
That change makes sense to me, but what about other test cases? can you attach the entire test log?

**Comment by ulwlu** [comment]
Thank you for reply. 
Yes, as I stated "First of all", I will check the rest as well after test_utils.

Here is the entire test log. 
The environment is the same as the one in description.

[test.log](https://github.com/pdm-project/pdm/files/6216796/test.log)

**Comment by ulwlu** [comment]
@frostming 
Hi. I just realized 12 out of 13 tests are all about accessing `demo-0.0.1.tar.gz` and `demo-0.0.1-py2.py3-none-any.whl`, and it failed with any kind of OS.

I did `pdm run test` on docker as well, and it occurs 12 errors like below.

![ss](https://user-images.githubusercontent.com/41639488/112745063-74cd7400-8fe0-11eb-9cb9-5bd9953c22e1.png)

Then, 1 out of 13 tests is fixed on PR, so the only problem about demo-data left.
